pkg/store/bucket.go

```diff
@@ -2151,6 +2156,53 @@ func (r *bucketIndexReader) reset() {
 // chunk where the series contains the matching label-value pair for a given block of data. Postings can be fetched by
 // single label name=value.
 func (r *bucketIndexReader) ExpandedPostings(ctx context.Context, ms []*labels.Matcher, bytesLimiter BytesLimiter) ([]storage.SeriesRef, error) {
+	// Sort matchers to make sure we generate the same cache key.
```

Comment:
Maybe we can move this new section into a function to reduce some nesting with early returns, e.g. `!hit` and `err != nil`.

Reply:
Updated with the new commit